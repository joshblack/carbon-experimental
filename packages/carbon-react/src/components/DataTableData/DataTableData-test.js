import React from 'react';
import DataTableData from '../DataTableData';
import { shallow } from 'enzyme';

describe('DataTableData', () => {
  it('should render', () => {
    const wrapper = shallow(
      <DataTableData>
        <div>Children</div>
      </DataTableData>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render children as expected', () => {
    const wrapper = shallow(
      <DataTableData>
        <div>Children</div>
      </DataTableData>
    );
    expect(wrapper.find('div').length).toEqual(1);
  });
});
