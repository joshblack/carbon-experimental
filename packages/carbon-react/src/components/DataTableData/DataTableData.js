import PropTypes from 'prop-types';
import React from 'react';
import Icon from '../Icon';
import classNames from 'classnames';

const DataTableData = props => {
  const {
    children,
    className,
    isSelectable,
    isSelected,
    value,
    id,
    isExpandable,
    isExpanded,
    ...other
  } = props;

  const selectableElement = (
    <span>
      <input
        checked={isSelected}
        id={id}
        className="bx--checkbox"
        type="checkbox"
        value={value}
        name={id}
      />
      <label htmlFor={id} className="bx--checkbox-label">
        <span className="bx--checkbox-appearance">
          <svg
            className="bx--checkbox-checkmark"
            width="12"
            height="9"
            viewBox="0 0 12 9"
            fillRule="evenodd">
            <path d="M4.1 6.1L1.4 3.4 0 4.9 4.1 9l7.6-7.6L10.3 0z" />
          </svg>
        </span>
      </label>
    </span>
  );

  const expandedElement = (
    <button className="bx--table-expand-v2__button">
      <Icon
        name="chevron--right"
        description="Expandable icon"
        className="bx--table-expand-v2__svg"
      />
    </button>
  );

  const tableDataClassnames = classNames(className, {
    'bx--table-expand-v2': isExpandable,
  });

  return (
    <td
      className={tableDataClassnames}
      {...other}
      data-previous-value={isExpanded ? 'collapsed' : ''}>
      {isSelectable && selectableElement}
      {isExpandable && expandedElement}
      {children}
    </td>
  );
};

DataTableData.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

export default DataTableData;
