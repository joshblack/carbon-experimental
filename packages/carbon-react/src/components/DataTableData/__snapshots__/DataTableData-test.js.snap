// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableData should render 1`] = `
<td
  className=""
  data-previous-value=""
>
  <div>
    Children
  </div>
</td>
`;
